src/main/java/skyler/Storage.java

```diff
@@ -0,0 +1,232 @@
+        String formattedTime = String.format("%s:%s", hour, minute);
+
+        LocalDate date = LocalDate.parse(timeInfo[0]);
+        LocalTime time = LocalTime.parse(formattedTime);
+        return LocalDateTime.of(date, time);
+    }
+
+    // load from file and return arraylist of tasks
+    // throws exception if folder / file does not exist (will create if so)
+
+    //method below adapted from
+    //https://www.geeksforgeeks.org/java-program-to-search-for-a-file-in-a-directory/ ,
+    //https://www.educba.com/java-directories/ and
+    //https://www.tutorialspoint.com/determine-if-file-or-directory-exists-in-java
+
+    /**
+     * Loads tasks from file and returns the corresponding list of tasks
+     *
+     * @return ArrayList of tasks previously stored in the file.
+     * @throws SkylerException If directory or file not found.
+     * @throws IOException If IO operation fails.
+     */
+    public ArrayList<Task> load() throws SkylerException, IOException {
+        ArrayList<Task> tasks = new ArrayList<>();
+
+        String[] dir = filePathChosen.split("/");
+        String file = dir[dir.length - 1];
+        String currPath = "";
+        for (int i = 0; i < dir.length - 1; i++) {
+            currPath += dir[i];
+            String name = new File(currPath).getAbsolutePath();
+            Path path = Paths.get(name);
+            // if folder does not exist yet
+            if (!Files.exists(path)) {
+                Files.createDirectory(path);
+            }
+            currPath += "/";
+        }
+
+        String filePath = new File(currPath).getAbsolutePath();
+        File directory = new File(filePath);
+
+        String[] flist = directory.list();
+        int flag = 0;
+        // directory is empty
+        if (flist == null) {
+            filePath = new File(currPath + file).getAbsolutePath();
+            directory = new File(filePath);
+            directory.createNewFile();
+            throw new SkylerException();
+        }
+
+        for (int i = 0; i < flist.length; i++) {
+            String filename = flist[i];
+            if (filename.equals(file)) {
+                filePath = new File(currPath + file).getAbsolutePath();
+                directory = new File(filePath);
+                flag = 1;
+            }
+        }
+
+        // directory is not empty but specified file not found
+        if (flag == 0) {
+            filePath = new File(currPath + file).getAbsolutePath();
+            directory = new File(filePath);
+            directory.createNewFile();
+            throw new SkylerException();
+        }
+
+        Scanner sc = new Scanner(directory);
+        while (sc.hasNextLine()) {
+            String taskStr = sc.nextLine();
+            String taskDescription = taskStr.substring(8);
+
+            char type = taskStr.charAt(0);
+            int taskStatus = Character.getNumericValue(taskStr.charAt(4));
+
+            switch(type) {
```

Comment:
Missing a whitespace here, easily fixed with the IDE configurations.
```suggestion
            switch (type) {
```